cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/management-credential-step/ManagementCredentials.tsx

```diff
@@ -127,48 +125,24 @@ function ManagementCredentials(props: Partial<StepProps>) {
             setMessage(`Unable to connect to AWS: ${err.body.message}`);
         }
     };
-    const resetField = (field: string) => {
-        if (handleValueChange && awsState[STORE_SECTION_FORM][field]) {
-            handleValueChange(INPUT_CHANGE, field, '', currentStep, errors);
-            setValue('EC2_KEY_PAIR', '');
-        }
-    };
-    const handleSelectProfile = (profile: string) => {
-        setConnectionStatus(CONNECTION_STATUS.DISCONNECTED);
-        resetField('EC2_KEY_PAIR');
-        if (handleValueChange) {
-            setTimeout(() => {
-                handleValueChange(INPUT_CHANGE, 'PROFILE', profile, currentStep, errors);
-            });
-        }
-    };
-
-    const handleSelectRegion = (region: string) => {
-        setConnectionStatus(CONNECTION_STATUS.DISCONNECTED);
-        resetField('EC2_KEY_PAIR');
-        if (handleValueChange) {
-            setTimeout(() => {
-                handleValueChange(INPUT_CHANGE, 'REGION', region, currentStep, errors);
-            });
-        }
-    };
 
     const handleSelectKeyPair = (event: ChangeEvent<HTMLSelectElement>) => {
-        setValue('EC2_KEY_PAIR', event.target.value, { shouldValidate: true });
-        if (handleValueChange) {
-            setTimeout(() => {
-                handleValueChange(INPUT_CHANGE, 'EC2_KEY_PAIR', event.target.value, currentStep, errors);
-            });
-        }
+        awsDispatch({
+            type: INPUT_CHANGE,
+            field: AWS_FIELDS.EC2_KEY_PAIR,
+            payload: event.target.value,
+        } as FormAction);
     };
 
-    const handleInputChange = (field: string, value: string) => {
+    const reset = () => {
```

Comment:
nit: maybe `resetEc2KeyPair` instead of `reset`?

Reply:
renamed